### ISSUE TITLE
Region filter UI implementation

### ISSUE THREAD

**Issue by jmunroe** [cioos-siooc/cioos-siooc-assetmap#28]
Currently the asset map only searches the CKAN database by keyword. We need add functionality to search by geographic region

Sub tasks:

1. Prototype a UI 'area selector' control. This could be a rectangular selector tool allowing the selection of the a region on the map or a polygon drawing tool. Can multiple areas be selected at the same time? Can a previously modified area be modified through the UI?

2. Prototype a UI tool to select 'default' regions.  Could be a dropbox control that is pre-populated with different the different regions (e.g. Pacific, Great Lakes, St. Lawrence, Arctic, Atlantic, etc.). This should be configured with a .json file in the resources/ subdirectory.

3. Modify the CKAN query to use this area selector(s) to restrict the number of results returned and displayed.

4. Add a method that allows the regional selection to be disabled.

**Comment by jmunroe** [comment]
I will ask @justhateme to work on this task. If there is already work started on this issue, please comment here so that we do not duplicate work.

If this task can be made more specific (especially based on discussion in the UX task team meeting next week, @GermainSauve @odumidika @fostermh) please provide direction here.

**Comment by jmunroe** [comment]
task: We can do geographical search using the CKAN  Web UI. How to do we do a geographic query through the CKAN API?

**Comment by jmunroe** [comment]
@NeZanyat has started work on this issue.

**Comment by NeZanyat** [comment]
@jmunroe Hello James, what projection should I use for spatial search on your CKAN instance?

**Comment by jmunroe** [comment]
Not sure... I assume it is EPSG:4326 but I'll try and confirm that.

**Comment by sauve** [comment]
From memory this is EPSG:4326, but I wonder if the metadata profile specify the projection as an attribute or if it forces EPSG:4326. If it doesn't force it, it should be reprojected for the CKAN schema. Wonder what @fostermh as to say?

**Comment by fostermh** [comment]
Hi @NeZanyat, @jmunroe, @sauve,
The ckan DB uses and srid  of 4326 by default. The spatial search defaults to 4326 as well. When performing a query (api call) you can specify a different srid by setting the crs parameter. for example to use wgs84 web mercator (auxiliary sphere) you would pass `crs=3857` as part of the url query string. CKAN will attempt to convert the data to 4326 using the ST_Transform method of postgis. 

So in summary, use 4326 if you can but anything will work as long as you specify it in crs.
